src/NetMQ/ReceivingSocketExtensions.cs

```diff
@@ -393,13 +370,9 @@ public static void ReceiveMessage([NotNull] this IReceivingSocket socket, [NotNu
         {
             message.Clear();
 
-            bool more = true;
-
+            var more = true;
             while (more)
-            {
-                byte[] buffer = socket.Receive(dontWait, out more);
-                message.Append(buffer);
-            }
+                message.Append(socket.Receive(dontWait, out more));
```

Comment:
Good spot. I've a complete overhaul of these methods coming in a subsequent
PR and will make sure this is included.

[via phone]
On 11 Mar 2015 08:09, "Doron Somech" <notifications@github.com> wrote:

> In src/NetMQ/ReceivingSocketExtensions.cs
> <https://github.com/zeromq/netmq/pull/292#discussion_r26194230>:
>
> >              while (more)
> > -            {
> > -                byte[] buffer = socket.Receive(dontWait, out more);
> > -                message.Append(buffer);
> > -            }
> > +                message.Append(socket.Receive(dontWait, out more));
>
> here we can call the method with the Msg parameter and improve performance
> a little by reusing the msg structure.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zeromq/netmq/pull/292/files#r26194230>.
>